workflows/detaxizer.nf

```diff
@@ -105,95 +121,175 @@ workflow DETAXIZER {
 
-    KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
 
-    ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN (
-        ch_combined
-    )
+        //
+        // MODULE: Parse the taxonomy from the kraken2 report and return all subclasses of the tax2filter
+        //
+        PARSE_KRAKEN2REPORT(
+            KRAKEN2_KRAKEN2.out.report.take(1)
+        )
+        ch_versions = ch_versions.mix(PARSE_KRAKEN2REPORT.out.versions)
 
-    ch_versions = ch_versions.mix(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.versions.first())
 
-    //
-    // MODULE: Summarize the kraken2 results and the isolated kraken2 hits
-    //
-    ch_prepare_summary_kraken2 = KRAKEN2_KRAKEN2.out.classified_reads_assignment.join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified).map {
-        meta, path1, path2 ->
-            return [ meta, [ path1, path2 ] ]
+        //
+        // MODULE: Isolate the hits for a certain taxa and subclasses
+        //
+        ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+
+        KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
+
+        ISOLATE_KRAKEN2_IDS (
+            ch_combined
+        )
+
+        ch_versions = ch_versions.mix(ISOLATE_KRAKEN2_IDS.out.versions.first())
+
     }
 
-    ch_combined_kraken2 = ch_prepare_summary_kraken2.map {
-        meta, path ->
-            return [ meta +[ id: meta.id.replaceAll("(_R1|_R2)", "") ] , path]
-        }
-        .map {
-            meta, path ->
-                path = path.flatten()
-                return [meta, path]
-            }
+    if (params.classification_bbduk) {
+
+
+        //
+        // MODULE: Run bbduk
+        //
+        BBMAP_BBDUK (
+            ch_fastq_for_classification,
+            fasta_bbduk.first()
+        )
+        ch_versions = ch_versions.mix(BBMAP_BBDUK.out.versions.first())
+
+
+        //
+        // MODULE: Run ISOLATE_BBDUK_IDS
+        //
+        ISOLATE_BBDUK_IDS(
+            BBMAP_BBDUK.out.contaminated_reads
+        )
+        ch_versions = ch_versions.mix(ISOLATE_BBDUK_IDS.out.versions.first())
+
+
+    }
 
-    ch_kraken2_summary = SUMMARY_KRAKEN2(
-        ch_combined_kraken2
+    // Prepare MERGE_IDS Channel (with or without merging of IDs)
+
+    if (params.classification_kraken2 || (!params.classification_kraken2 && !params.classification_bbduk)){
+
+
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_KRAKEN2_IDS.out.classified_ids
         )
 
-    ch_versions = ch_versions.mix(ch_kraken2_summary.versions.first())
+
+    } else if (params.classification_bbduk && !params.classification_kraken2){
+
+
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_BBDUK_IDS.out.classified_ids
+        )
+
+
+    } else if (params.classification_bbduk && params.classification_kraken2){
```

Comment:
That looks to me as if it would never be executed, because `(params.classification_kraken2 && params.classification_bbduk)` would already be true for `if (params.classification_kraken2 || (!params.classification_kraken2 && !params.classification_bbduk))`? I.e. shift this part here to the top.